### PR TITLE
quincy: mgr/cephadm: Adding extra arguments support for RGW frontend

### PR DESCRIPTION
**Pull request by adk3798**
Backport of both https://github.com/ceph/ceph/pull/48630 and https://github.com/ceph/ceph/pull/52301 which fixed a bug from the first PR.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/cephadm/services/rgw.rst [hunk 1]
```diff
@@ -74,6 +74,33 @@ example spec file:
     spec:
       rgw_frontend_port: 8080
 
+Passing Frontend Extra Arguments
+--------------------------------
+
+The RGW service specification can be used to pass extra arguments to the rgw frontend by using
```
Comment (anthonyeleven) [inline doc/cephadm/services/rgw.rst:80]: s/rgw/RGW/

File: doc/cephadm/services/rgw.rst [hunk 2]
```diff
@@ -74,6 +74,33 @@ example spec file:
...
+      rgw_realm: myrealm
+      rgw_zone: myzone
+      rgw_frontend_type: "beast"
+      rgw_frontend_port: 5000
+      rgw_frontend_extra_args:
+      - "tcp_nodelay=1"
+      - "max_header_size=65536"
+
+.. note:: cephadm combines the arguments from the `spec` section and the ones from
+	  the `rgw_frontend_extra_args` into a single space-separated arguments list
```
Comment (anthonyeleven) [inline doc/cephadm/services/rgw.rst:102]: s/the//

File: doc/cephadm/services/rgw.rst (hunk 1 above)
Comment (adk3798) [reply doc/cephadm/services/rgw.rst:80]: While this is true, this is a quincy backport so if we'd like to change it we should probably have a patch in main and then propagate it down. Same goes for the other comment.

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by adk3798** [comment]
jenkins test api

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-03-11_04:43:35-orch:cephadm-wip-adk2-testing-2024-03-10-1540-quincy-distro-default-smithi/

reruns: https://pulpito.ceph.com/adking-2024-03-11_12:11:09-orch:cephadm-wip-adk2-testing-2024-03-10-1540-quincy-distro-default-smithi/

After reruns, 2 failures

- `test_non_existent_cluster` failure is caused by a change to the `ceph nfs cluster info` command to make it return a non-zero return code when the cluster does not exist. The change itself was backported, but the change to make the test expect that return code was not.
- `rcu: INFO: rcu_sched detected stalls on CPUs/tasks: ' in syslog` in mgr-nfs-upgrade test, known issue

Nothing to block merging

**Pull request merged**